### DISCUSSION TITLE
Payload variables

### DISCUSSION THREAD

**Discussion by jakes670** [jaredhendrickson13/pfsense-api#304] [Q&A]
I am new to this, please bear with me. 

I am creating a LDAP server with the API, there is one variable.  Where can I get the list of variables that I can use in the API call

eg. 

I want to set the "RFC 2307 Groups" tik mark to off under the LDAP endpoint /api/v1/user/auth_server/ldap endpoint

I have tried a few below none is working
                "ldap_rfc_2307_enabled": False,
                "rfc2307_groups_enabled": False,
                "ldap_rfc2307_groups": False,
                "ldap_rfc_2307_enabled": False,
            
Thanks

**Comment by jaredhendrickson13** [comment]
Hey!

You can view the available API properties within the API documentation. After installing the API, following these steps:

1) Login to the pfSense webConfigurator and navigate to System > API > Documentation
2) Locate the endpoint you wish to use (/api/v1/user/auth_server/ldap in your case) and expand the endpoint documentation.
3) You should see the 'Example Value' in the expanded endpoint documentation, next to this item you should see `Schema`. Click the `Schema' button to show the available properties, their typing, and descriptions.

As for your specific issue, the `ldap_rfc2307` property is currently coupled with `ldap_attr_groupobj` so simply specifying any value in `ldap_attr_groupobj` will enable `ldap_rfc2307`. 

Let me know if you have any issues.

Thanks!

**Comment by jakes670** [comment]
Thanks Jared

My problem is that value get selected without specifying anything.

How can I unselect the value, I have tried the below.

"ldap_attr_groupobj ": "",
But the value is selected with and without it.

Here is my post

post_tests = [
        {
            "name": "Create LDAP auth server",
            "payload": {
                "name": "LDAP",
                "host": "ldap.server",
                "ldap_port": 636,
                "ldap_urltype": "encrypted",
                "ldap_protver": "3",
                "ldap_timeout": 25,
                "ldap_scope": "subtree",
                "ldap_basedn": "o=server,dc=acme,dc=com",
                "ldap_authcn": "ou=Users,o=server,dc=acme,dc=com",
                "ldap_extended_enabled": True,
                "ldap_extended_query": "&(objectClass=inetOrgPerson)(uid=*)",
                "ldap_attr_groupobj ": "",
                "ldap_attr_user": "uid",
                "ldap_attr_group": "cn",
                "ldap_attr_member": "memberOf",
                "ldap_attr_groupobj": "groupOfNames",
                "ldap_binddn": "uid=bind,ou=Users,o=server,dc=acme,dc=com",
                "ldap_bindpw": "passw",
               "active": False
           }
        }

Thanks
Jaco

**Comment by jaredhendrickson13** [comment]
In your example above it looks like `ldap_attr_groupobj` is being specified twice. You will need to remove `ldap_attr_groupobj` from the request data altogether. As an example, the following request would create an LDAP authentication server without `ldap_rfc2307` being enabled:

```json
{
    "name": "EXAMPLE",
    "host": "ldap.example.com",
    "ldap_port": 636,
    "ldap_urltype": "encrypted",
    "ldap_protver": "3",
    "ldap_timeout": 12,
    "ldap_scope": "subtree",
    "ldap_basedn": "dc= example,dc=com",
    "ldap_authcn": "ou=people,dc= example,dc=com;ou=groups,dc= example,dc=com",
    "ldap_attr_user": "uid",
    "ldap_attr_group": "cn",
    "ldap_attr_member": "memberof",
    "ldap_binddn": "cn=pfsense,dc= example,dc=com",
    "ldap_bindpw": "example",
    "active": false
}
```

If you are looking to disable `ldap_rfc2307` but still pass in a value for `ldap_attr_groupobj`, that would not be possible currently. In that case, you are welcome to [open an issue](https://github.com/jaredhendrickson13/pfsense-api/issues/new/choose) to request decoupling those two properties from each other and that can likely be changed in the next patch.
